core/src/main/java/com/linecorp/armeria/internal/server/annotation/DefaultAnnotatedService.java

```diff
@@ -220,40 +223,53 @@ private static void warnIfHttpResponseArgumentExists(Type returnType,
         }
     }
 
-    public String serviceName() {
+    @Override
+    public String name() {
+        if (serviceName == null) {
+            return serviceClass().getName();
+        }
         return serviceName;
     }
 
     public boolean serviceNameSetByAnnotation() {
         return serviceNameSetByAnnotation;
     }
```

Comment:
IIRC @ikhoon wanted to remove this. Could you give some detailed ideas to @chickenchickenlove, @ikhoon ?

Reply:
I understand that he want to remove `serviceNameSetByAnnotation` from `AnnotatedService`. (public API). 
Maybe, i don't understand his Intent well 😢

Reply:
I removed this method! I think it is not needed anymore. 
Please refer to this comments! (https://github.com/line/armeria/pull/5628#discussion_r1577345982).

Reply:
Looks good!